lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs

```diff
@@ -494,36 +499,81 @@ monitorStakePools tr gp nl db@DBLayer{..} = do
     mkLatestGarbageCollectionEpochRef = newIORef minBound
 
     initCursor :: IO [BlockHeader]
-    initCursor = atomically $ readPoolProductionCursor (max 100 k)
+    initCursor = do
+        fromDB <- atomically $ readPoolProductionCursor (max 100 k)
+        pure (lastByronBlock:fromDB)
       where k = fromIntegral $ getQuantity getEpochStability
+            -- We only have stake pools since shelley era.
+            -- This is hardcoded for mainnet and will do nothing for other
+            -- networks.
+            -- TODO: query cardano-node for hard fork block (not yet possible)
+            --       https://jira.iohk.io/browse/CAD-1850
+            lastByronBlock :: BlockHeader
+            lastByronBlock = BlockHeader
+                (SlotNo 4492799)
+                (Quantity 4490510)
+                (Hash $ unsafeFromHex
+                    "f8084c61b6a238acec985b59310b6ecec49c0ab8352249afd7268da5cff2a457")
+                (Hash $ unsafeFromHex
+                    "aa83acbf5904c0edfe4d79b3689d3d00fcfc553cf360fd2229b98d464c28e9de")
```

Comment:
I still think this would be better as a top-level constant.

Embedding this as a nested constant means that it's doesn't fit within our [line limit](https://github.com/input-output-hk/cardano-wallet/wiki/Coding-Standards#limit-line-length-to-80-characters).

But by moving this to the top-level, we can completely avoid this problem.

As a bonus, we can add a Haddock comment:

```suggestion
-- | The very last block header of the Byron era, just before the transition
--   to Shelley.
-- 
-- This is hard-coded for mainnet and will do nothing for other networks.
--
-- TODO: query cardano-node for hard fork block (not yet possible).
-- https://jira.iohk.io/browse/CAD-1850
--
lastByronBlock :: BlockHeader
lastByronBlock = BlockHeader
    (SlotNo 4492799)
    (Quantity 4490510)
    (Hash $ unsafeFromHex
        "f8084c61b6a238acec985b59310b6ecec49c0ab8352249afd7268da5cff2a457")
    (Hash $ unsafeFromHex
        "aa83acbf5904c0edfe4d79b3689d3d00fcfc553cf360fd2229b98d464c28e9de")
```